### ISSUE TITLE
`setTimeout()` doesn't fire unless the Qt event loop is active

### ISSUE THREAD

**Issue by sedwards2009** [nodegui/node#4]
I just noticed that `setTimeout()` doesn't fire unless there is activity on the Qt event loop.

This sounds very similar to the problem reported here: https://github.com/nodegui/nodegui/issues/864

**Comment by sedwards2009** [comment]
I am using the 16.4.1 version.

**Comment by sedwards2009** [comment]
Also, so far I've only noticed this problem on Windows.

**Comment by J-Cake** [comment]
I'm having a similar issue relating to using any sort of Promise-based operations. (presumably this is an issue with Qode modifying the underlying event loop)

My application closes gracefully before the resolution of a promise-based timeout of roughly 1000ms. For example

```typescript
import timers from 'node:timers/promises';
import * as ng from '@nodegui/nodegui';

const win = new ng.QMainWindow();

win.show();

console.log('Get Here');
const timeout = await timers.setTimeout(1000);
console.log('Not Getting Here'); // This section is never reached. 
```

My guess is that Qode assumes the event queue is clear and proceeds to exit the application. What I don't understand is that if Qode synchronises two event loops (as per my understanding), the LibUV event queue and the Qt event loop, the window still remains open, and so `.show()` should not return. 

The annoying thing is that I'm relying heavily on slow operations for which I use promises. The application closes before any real work can be done.

**Comment by sedwards2009** [comment]
I also noticed that qode/NodeGui can get into a state where my app (a terminal) becomes very slow and sluggish. I strongly suspect this has to do with I/O events not being processed in a timely way.

I've spent some time debugging into this, and have found a couple interesting things so far which may be enough to explain the problem:
* Sometimes `PostMessage()` from the monitoring thread to the main thread (actually window) can fail. When this happens, the queue on the window is completely full and "over quota".
* The semaphore being used to coordinate the monitor thread and running the nodejs event processing on the main thread, gets hits by `sem_post()` many many times if you are doing I/O before the Qt event loop is entered. This causes the counter in the semaphore to be huge instead of 0 or 1. This gives the monitor thread the ability to spam `PostMessage()` in a tight loop
 
I've got some ideas which may fix these problems and my general problem, even if I don't quite figure out the exact details of how the symptoms appear.
